Tribler/community/bartercast4/community.py

```diff
@@ -133,17 +133,18 @@ def on_stats_response(self, messages):
     # bartercast accounting stuff
     def backup_bartercast_statistics(self):
         self._logger.debug("merging bartercast statistics")
-        _barter_statistics.persist(self, 1)
+        _barter_statistics.persist(self, self._dispersy)
 
     def unload_community(self):
         self._logger.debug("unloading the Barter4 community")
-        super(BarterCommunity, self).unload_community()
-
-        # store all cached statistics
-        self.backup_bartercast_statistics()
+        # store last cached statistics
+        # this screws up for some reason so disable for now...
+        # we lose only a small part (max 2 minutes) of statistics but should be fixed
```

Comment:
I would make this a TODO entry so we can easily grep for it. Other than that it looks fine to me.